src/test/ui/cfg-rustdoc.rs

```diff
@@ -0,0 +1,6 @@
+// compile-pass
```

Comment:
I wonder if this sample should use `#[deny(dead_code)]` (or whatever warning trips on unused structs) to ensure that regular compilation doesn't even notice the struct.

Is there an existing test that ensures that item with `cfg(rustdoc)` appear in documentation? I couldn't find any on a quick scan - it doesn't seem like the `doc(cfg)` tests use it. We should make sure that we have a test for it.

Reply:
We have one in `src/test/ui/cfg-rustdoc.rs`. :)

Reply:
That's this file that you're adding in this PR. It doesn't test that an item appears in documentation, only that it doesn't break compilation. I'm not sure this test is actually doing anything useful, even - in fact, removing the `#[cfg(rustdoc)]` from the struct doesn't change anything about the output. Can you change this test so that `main` tries to use `Foo` and capture the compilation error? And please add a `rustdoc` test that uses `cfg(any(windows, rustdoc))` and ensures the item is present when running on Unix.